File: src/main/java/com/ByTech/ByTech/Detail/models/Detail.java
package com.ByTech.ByTech.Detail.models;

import com.ByTech.ByTech.Product.models.Product;
import com.ByTech.ByTech.Sales.models.Sale;
import jakarta.persistence.*;

import java.util.Date;

@Entity
public class Detail {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @ManyToOne(optional = false, cascade = CascadeType.DETACH, fetch = FetchType.EAGER)

    private Product product;
    @ManyToOne(optional = false, cascade = CascadeType.DETACH, fetch = FetchType.EAGER)

    private Sale sale;

    private int amount;
    @Column(columnDefinition = "DATE")
    private Date date;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Product getProduct() {
        return product;
    }

    public void setProduct(Product product) {
        this.product = product;
    }

    public Sale getSale() {
        return sale;
    }

    public void setSale(Sale sale) {
        this.sale = sale;
    }

    public int getAmount() {
        return amount;
    }

    public void setAmount(int amount) {
        this.amount = amount;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

}

File: Movil/backend/ByTech_movil/src/main/kotlin/com/example/ByTech_movil/Detail/models/Detail.kt
package com.example.ByTech_movil.Detail.models

import com.example.ByTech_movil.Product.models.Product
import com.example.ByTech_movil.Sales.models.Sale
import jakarta.persistence.*
import java.util.Date

@Entity
class Detail {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    @ManyToOne(optional = false, cascade = arrayOf(CascadeType.DETACH), fetch = FetchType.EAGER)
    var product: Product? = null

    @ManyToOne(optional = false, cascade = arrayOf(CascadeType.DETACH), fetch = FetchType.EAGER)
    var sale: Sale? = null

    var amount: Int = 0

    @Column(columnDefinition = "DATE")
    var date: Date? = null
}
